include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.cpp

```diff
@@ -460,7 +460,9 @@ TimeIntBDF<dim, Number>::postprocessing() const
   if(this->param.ale_formulation and this->get_time_step_number() == 1 and
      not this->param.restarted_simulation)
   {
-    operator_base->move_grid_and_update_dependent_data_structures(this->get_time());
+    operator_base->move_grid(this->get_time());
+    operator_base->update_matrix_free_after_grid_motion();
+    operator_base->update_spatial_operators_after_grid_motion();
```

Comment:
the sequence of these steps is important ... do we or _should we_ enforce the correct sequence of steps? One could work with `bool`s and assert invalid combinations. On the other side it is highlighted in the comments in `spatial_operator_base`.

Reply:
Yes, good point. The problem in my opinion is that the sequence of these three steps is neither the responsibility of the `SpatialOperatorBase` (because the `GridMotion` and `MatrixFree` objects are not owned by the spatial operator) nor of the `TimeIntegrator` (because the `GridMotion` and `MatrixFree` objects are not owned by the time integrator). For this reason, I suggest to keep it as is for now and address this in upcoming PRs in combination with other refactorings. One could of course introduce an additional function here or there, but this means that the interface is duplicated with the function `move_grid()` and needs to be adapated at two places when introducing new parameters (such as the update of a preconditioner, which was the origin of the present PR ...).

Overall, we are somewhat in conflict in terms of a separation of concerns between `SpatialOperator`/`Driver`/`TimeIntegrator` regarding data structures such as `Grid`/`GridMotion`/`Mapping`/`MatrixFree`. This is a somewhat everlasting topic in ExaDG where we still need to improve the design.

I think it makes sense that the `Driver` owns most of these data structures, because this allows us great flexibility in terms of coupled or multiphysics problems. One solution could be to pass a lambda with the sequence of three steps from the `Driver` (where all the relevant data structures live) to the `TimeIntegrator`, which can then call this lambda without any knowledge of `GridMotion`/`Mapping`/`MatrixFree` (data structures that should actually not occur in the time integration class).

Reply:
I will dig into the grid mapping now anyways, then I can better understnad the problem. For now, I think we can proceed with the three consecutive calls! Once I have the full picture, we can go for some refactoring :100: .